### ISSUE TITLE
Current enum library in use is 1.7.7, causing errors when PHP version is 8.1+

### ISSUE THREAD

**Issue by anastas10s-afk** [Automattic/woocommerce-payments#5533]
### Describe the bug

[The message/error can be seen here](https://gist.github.com/anastas10s-afk/a92eda6a03e0567c362b5899afc278e0), literally overtaking the whole screen.

### To Reproduce

Suffices to have WooCommerce & WooCommerce Payments active on a site, where the PHP version is 8.1+, for the errors to start appearing.

### Actual behavior

WooCommerce Payments apparently [returns an error, with PHP versions 8.1.14 & 8.2.1](https://wordpress.org/support/topic/status-on-updating-depreciated-php-call-please/#post-16389912), in my staging installation.
[Nothing I could find on GitHub](https://github.com/Automattic/woocommerce-payments/issues?q=is%3Aissue+in%3Atitle+php+is%3Aopen).

### Screenshots

https://snipboard.io/vjslMy.jpg

### Expected behavior

Have the enum library updated, please.
The enum library currently in use is 1.7.7: 
https://github.com/Automattic/woocommerce-payments/blob/develop/composer.lock

While its newer versions contain a fix: 
https://github.com/myclabs/php-enum/releases

### Additional context

- https://wordpress.org/support/topic/error-with-php-8-1/

**Comment by csmcneill** [comment]
Hey @anastas10s-afk! Thanks for creating this issue!

I reviewed the thread on WordPress.org, but I'm unable to reproduce the issue when using other versions of PHP 8.1.x. I have a few questions:

1. Does this issue _only_ occur with specific versions of PHP (e.g. 8.1.14 and 8.2.1)? 
2. Does `WP_DEBUG_DISPLAY` need to be set to `true` for these deprecation notices to be present?
3. If you use a different version of PHP 8 (e.g. 8.1.15 or 8.2.2), are the deprecation notices still present?

Looking forward to hearing from you!

**Comment by anastas10s-afk** [comment]
Hey @csmcneill 👋

Thank you for checking this further. In response to your questions:

1. Yes, indeed. With PHP version 8.0.27 it does not.
2. Yes, correct. I just verified this. When switched to "false", I can see nothing. Here is a screenshot showcasing the settings in that section: https://snipboard.io/Xh9QRK.jpg
3. Provided that `define( 'WP_DEBUG', true );`, then, yes. I just tried with PHP version 8.1.15.

I hope that helps!
